lib/reverse_sentence.rb

```diff
@@ -1,6 +1,80 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n): there are two loops, first loop is O(n/2) and the second loop is O(n).
+#                        So, O(n/2) + O(n) --> O(n)
+# Space complexity: O(1): In the first loop on line 16, it reverses the string in place.
+#                         In the second loop on line 24, the helper method "reverse_a_word" reverses each word in place,
+#                         and the reversed word replaces the original String[x..y], it's in place replacement.
+#                         Plus extra variables, overall, the space complexity is a constant.
+
 def reverse_sentence(my_sentence)
```

Comment:
👍  This works, just note that you're reversing twice here.  Could you make a helper method for this?

Reply:
Ohoh! Just realized how redundant it was...orz  I replaced the reverse code at step1 by using the helper method to do the reverse.